Parser/Device/AbstractDeviceParser.php

```diff
@@ -1861,6 +1861,16 @@ public function parse(): ?array
         $resultClientHint = $this->parseClientHints();
         $deviceModel      = $resultClientHint['model'] ?? '';
 
+        // is freeze user-agent then restoring the original UA for the device definition
+        if ('' !== $deviceModel && \preg_match('~Android 10[.\d]*; K(?: Build/|[;)])~i', $this->userAgent)) {
+            $osVersion = $this->clientHints->getOperatingSystemVersion();
+            $this->setUserAgent(\preg_replace(
+                '(Android 10[.\d]*; K)',
+                \sprintf('Android %s; %s', $osVersion, $deviceModel),
+                $this->userAgent
+            ));
```

Comment:
In theory it could happen that a device model was provided through the client hints, but the os version is empty. This would in that case end up in a user agent where no version is set at all. Maybe we should keep Android 10 if no version is provided?

Reply:
We have rules based on the version + code of the device. I will add a check for the presence of OS

Reply:
@sgiehl You can watch this PR again? I made changes.